### PR TITLE
chore(doc): update developers doc

### LINKED ISSUE #5464: Make images does not work out of the box
### What happened?

The `make images` does not work out of the box as it requires `docker-buildx` to be installed, but it is not mentioned in the developer/contribution guide. If you install` docker-buildx,` them the build succeed. 

### Steps to reproduce

1. checkout the project
2. follow the instruction on https://camel.apache.org/camel-k/next/contributing/developers.html
3. run make images
4. it should fail with the provided logs

### Relevant log output

```shell
####### Building Camel K operator arch amd64 container image...
mkdir -p build/_maven_output
docker buildx build --target base --platform=linux/amd64 --build-arg IMAGE_ARCH=amd64 --load -t docker.io/apache/camel-k:2.4.0-SNAPSHOT-amd64 -f build/Dockerfile .
unknown flag: --target
See 'docker --help'.

Usage:  docker [OPTIONS] COMMAND

A self-sufficient runtime for containers

Common Commands:
  run         Create and run a new container from an image
  exec        Execute a command in a running container
  ps          List containers
  build       Build an image from a Dockerfile
  pull        Download an image from a registry
  push        Upload an image to a registry
  images      List images
  login       Log in to a registry
  logout      Log out from a registry
  search      Search Docker Hub for images
  version     Show the Docker version information
  info        Display system-wide information

Management Commands:
  builder     Manage builds
  container   Manage containers
  context     Manage contexts
  image       Manage images
  manifest    Manage Docker image manifests and manifest lists
  network     Manage networks
  plugin      Manage plugins
  system      Manage Docker
  trust       Manage trust on Docker images
  volume      Manage volumes
...
```
```


### Camel K version

main

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #5464

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/contributing/developers.adoc [hunk 1]
```diff
@@ -22,18 +22,17 @@ There are many ways you can contribute to Camel K, not only software development
...
 * **JDK version 17+**: the build requires JDK version 17 or above. This corresponds to the JDK version of the integration base image.
 * **Maven version 3.8+**: the build requires Maven 3.8 or above. This corresponds to the version defined in the `build/Dockerfile`.
 * **MinGW**: needed to compile the project on Windows. Refer to the https://www.mingw-w64.org/[MinGW website] for the installation.
 * **Windows Subsystem for Linux (WSL)**: for running Linux binary executables natively on Windows. Refer to https://docs.microsoft.com/en-us/windows/wsl/install[WSL Website] for installation. Alternatively, you can use https://www.cygwin.com/[Cygwin] or https://www.educative.io/edpresso/how-to-install-git-bash-in-windows[Git Bash].
-
-NOTE: MacOS users will need to use *gnu-sed* to successfully run the Make build scripts (e.g. for generating the Camel K bundle). Please install gnu-sed on your machine (e.g. `brew install gnu-sed`) and set your PATH accordingly to use gnu-sed with: `export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"`
```
Comment (squakez) [inline docs/modules/ROOT/pages/contributing/developers.adoc:32]: Any reason to remove this part? Is it no longer relevant?

File: docs/modules/ROOT/pages/contributing/developers.adoc [hunk 2]
```diff
@@ -22,18 +22,17 @@ There are many ways you can contribute to Camel K, not only software development
 
 In order to build the project, you need to comply with the following requirements:
 
-* **Go version 1.16+**: needed to compile and test the project. Refer to the https://golang.org/[Go website] for the installation.
+* **Go version 1.21+**: needed to compile and test the project. Refer to the https://golang.org/[Go website] for the installation.
```
Comment (squakez) [inline docs/modules/ROOT/pages/contributing/developers.adoc:25]: We could try to automate and scan the go.mod to get such info during nightly tasks.

**Review by squakez — APPROVED** [review]

File: docs/modules/ROOT/pages/contributing/developers.adoc (hunk 1 above)
Comment (lburgazzoli) [reply docs/modules/ROOT/pages/contributing/developers.adoc:32]: seems a mistake, let me bring it back

Comment (lburgazzoli) [reply docs/modules/ROOT/pages/contributing/developers.adoc:32]: done

**Review by squakez — APPROVED** [review]

**Pull request merged**